versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/AbstractMergeTransplant.java

```diff
@@ -148,14 +166,15 @@ void apply(
         throws Exception;
   }
 
-  private Hash[] mergeTransplant(MergeOrTransplant mergeOrTransplant) throws Exception {
+  private Hash[] mergeTransplant(int numCommits, MergeOrTransplant mergeOrTransplant)
+      throws Exception {
     BranchName main = BranchName.of("main");
     BranchName branch = BranchName.of("branch");
     BranchName conflict = BranchName.of("conflict");
 
     databaseAdapter.create(branch, databaseAdapter.hashOnReference(main, Optional.empty()));
 
-    Hash[] commits = new Hash[3];
+    Hash[] commits = new Hash[numCommits];
```

Comment:
Since the prod code change touches the handling of content-keys/values, it's appropriate to add assertions for the merged content-key/values on every merged/tranplanted commit. Should also have some delete-operations in the generated commits.

Reply:
Added validation and delete commit